### PR TITLE
api: handle moving file to another package

### PR DESCRIPTION
**Pull request by lizan**
Description:
Depends on https://github.com/cncf/udpa/pull/20

Risk Level:
Testing:
Docs Changes:
Release Notes:

### CODE REVIEW

File: tools/type_whisperer/api_type_db.proto [hunk 1]
```diff
@@ -21,9 +29,13 @@ message TypeDb {
 
   // Map from fully qualified package to the package's name in the next major
   // version of the API.
-  map<string, string> next_version_packages = 2;
+  // map<string, string> next_version_packages = 2;
+
+  // Map from .proto path to the proto's path in the next major version of the
+  // API.
+  // map<string, string> next_version_proto_paths = 3;
```
Comment (htuch) [inline tools/type_whisperer/api_type_db.proto]: Should we clean these up if no longer in use?

File: tools/type_whisperer/types.proto [hunk 2]
```diff
@@ -16,6 +16,9 @@ message TypeDescription {
   // Does this type have a breaking change? If so, it needs to be upgraded at
   // the next major version.
   bool next_version_upgrade = 5;
+
```
Comment (htuch) [inline tools/type_whisperer/types.proto]: Would it be possible to add a protoxform golden proto test or API type DB test to verify everything works as expected?

I'm thinking that the API boosting tool is going to be largely agnostic to what this PR does, providing everything is correct in the API type database, which is great.

File: tools/type_whisperer/api_type_db.proto (hunk 1 above)
Comment (lizan) [reply tools/type_whisperer/api_type_db.proto]: done

File: tools/type_whisperer/types.proto (hunk 2 above)
Comment (lizan) [reply tools/type_whisperer/types.proto]: yes added protoxform golden.

File: tools/proto_format.sh [hunk 3]
```diff
@@ -8,7 +8,7 @@ set -e
 
 # Clean up any stale files in the API tree output. Bazel remembers valid cached
 # files still.
-rm -rf bazel-bin/external/envoy_api
+# rm -rf bazel-bin/external/envoy_api
```
Comment (htuch) [inline tools/proto_format.sh]: ?

**Pull request merged**